src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = {};
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const dateArray = date.split(fromSeparator);
+
+  fromFormat.slice(0, -1).forEach((part, index) => {
+    dateParts[part] = dateArray[index];
+  });
+
+  // Перетворення року, якщо потрібно
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+    delete dateParts['YYYY'];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(dateParts['YY'], 10);
```

Comment:
Ensure that `dateParts['YY']` is defined before parsing it. If the input date does not contain a 'YY' part, this could lead to an error.